raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -587,31 +591,31 @@ contract TokenNetwork is Utils {
 
         total_deposit = participant1_state.deposit + participant2_state.deposit;
 
-        // The provided addresses must be the same as the recovered ones
-        require(participant1 == participant1_address);
-        require(participant2 == participant2_address);
-
-        // The channel must be open
-        require(channel.state == 1);
-
-        // The sum of the provided balances must be equal to the total deposit
-        require(total_deposit == (participant1_balance + participant2_balance));
-
         // Remove channel data from storage before doing the token transfers
         delete channel.participants[participant1];
         delete channel.participants[participant2];
         delete channels[channel_identifier];
 
+        emit ChannelSettled(channel_identifier);
+
         // Do the token transfers
         if (participant1_balance > 0) {
-            require(token.transfer(participant1, participant1_balance));
+            assert(token.transfer(participant1, participant1_balance));
         }
 
         if (participant2_balance > 0) {
-            require(token.transfer(participant2, participant2_balance));
+            assert(token.transfer(participant2, participant2_balance));
         }
 
-        emit ChannelSettled(channel_identifier);
+        // The channel must be open
+        assert(initial_state == 1);
```

Comment:
should be a require() as it depends on external input (time of call of the function)